core/src/test/scala/akka/persistence/jdbc/query/JournalDaoStreamMessagesMemoryTest.scala

```diff
@@ -0,0 +1,133 @@
+        val memoryLimit = 128000000
+        if (maxMem > memoryLimit)
+          throw new RuntimeException(
+            s"This test can only be run with a limited amount of memory ($memoryLimit), found [$maxMem]")
+
+        val payloadSize = 5000 // 5000 bytes
+        val eventsPerBatch = 1000
+
+        val numberOfInsertBatches = {
+          // calculate the number of batches using a factor to make sure we go a little bit over the limit
+          (maxMem.toInt / (payloadSize * eventsPerBatch) * 1.2).round.toInt
+        }
+        val totalMessages = numberOfInsertBatches * eventsPerBatch
+        val totalMessagePayload = totalMessages * payloadSize
+        log.info(
+          s"maxMem: $maxMem, batches: $numberOfInsertBatches (with $eventsPerBatch events), total messages: $totalMessages, total msgs size: $totalMessagePayload")
+
+        // payload can be the same when inserting to avoid unnecessary memory usage
+        val payload = Array.fill(payloadSize)('a'.toByte)
+
+        val lastInsert =
+          Source
+            .fromIterator(() => (1 to numberOfInsertBatches).toIterator)
+            .mapAsync(1) { i =>
+              val end = i * eventsPerBatch
+              val start = end - (eventsPerBatch - 1)
+              log.info(s"batch $i - events from $start to $end")
+              val atomicWrites =
+                (start to end).map { j =>
+                  AtomicWrite(immutable.Seq(PersistentRepr(payload, j, persistenceId)))
+                }.toSeq
+
+              dao.asyncWriteMessages(atomicWrites).map(_ => i)
+            }
+            .runWith(Sink.last)
+
+        // wait until we write all messages
+        // being very generous, 1 second per message
+        lastInsert.futureValue(Timeout(totalMessages.seconds))
+
+        val messagesSrc = dao.messages(persistenceId, 0, totalMessages, totalMessages)
+        val doneFut =
+          messagesSrc.runForeach {
+            case Success(repr) =>
+              log.info(s"fetched: ${repr.persistenceId} - ${repr.sequenceNr}/${totalMessages}")
+            case Failure(exception) => println(exception)
+          }
+
+        // wait until we read all messages
+        // being very generous, 1 second per message
+        doneFut.futureValue(Timeout(totalMessages.seconds))
+      }
+    }
+  }
+}
+
+class PostgresJournalDaoStreamMessagesMemoryTest
+    extends JournalDaoStreamMessagesMemoryTest("postgres-application.conf")
+    with PostgresCleaner
+
+// TODO: MySQL not working with withStatementParameters
+//class MySQLJournalDaoStreamMessagesMemoryTest
+//  extends ReplayPersistenceActorMemoryTest("mysql-application.conf")
+//    with MysqlCleaner
+
+class OracleJournalDaoStreamMessagesMemoryTest
+    extends JournalDaoStreamMessagesMemoryTest("oracle-application.conf")
+    with OracleCleaner
+
+class SqlServerJournalDaoStreamMessagesMemoryTest
+    extends JournalDaoStreamMessagesMemoryTest("sqlserver-application.conf")
+    with SqlServerCleaner
```

Comment:
Oracle and SqlServer works out-of-the-box (also without withStatementParameters). The driver don't keep things in memory.